src/qml/components/StorageLocations.qml

```diff
@@ -21,25 +21,28 @@ ColumnLayout {
         ButtonGroup.group: group
         text: qsTr("Default")
         description: qsTr("Your application directory.")
+        customDir: optionsModel.getDefaultDataDirString
         recommended: true
         checked: true
     }
     OptionButton {
+        id: customDirOption
         Layout.fillWidth: true
         ButtonGroup.group: group
         text: qsTr("Custom")
         description: qsTr("Choose the directory and storage device.")
+        customDir: fileDialog.folder
         onClicked: fileDialog.open()
     }
     FileDialog {
         id: fileDialog
         selectFolder: true
-        folder: optionsModel.getDefaultDataDirectory
+        folder: shortcuts.home
         onAccepted: {
             optionsModel.setCustomDataDirString(fileDialog.fileUrls[0].toString())
             var customDataDir = fileDialog.fileUrl.toString();
             if (customDataDir !== "") {
-                optionsModel.setCustomDataDirArgs(customDataDir);
+                customDirOption.customDir = optionsModel.setCustomDataDirArgs(customDataDir);
             }
         }
         onRejected: {
```

Comment:
Part of suggestion (2), when users cancels the file dialog, depending if the users changed the folder (stays in custom) or not (goes back to default).

```suggestion
        onRejected: {
            console.log("Custom datadir selection canceled")
            
            // if the user cancels to set a custom dir from the file dialog
            // default custom datadir check should be ticked
            if (fileDialog.folder == shortcuts.home) { 
                defaultDirOption.checked = true
            } else {
                // unless user accepted first another custom folder and then 
                // going back and forth from default cancels changing custom
                customDirOption.customDir = optionsModel.setCustomDataDirArgs(fileDialog.folder);
            }
```

Reply:
also great suggestion